app/lib/account/agent.dart

```diff
@@ -111,9 +122,14 @@ class AuthenticatedGithubAction implements AuthenticatedAgent {
 }
 
 /// Holds the authenticated Google Cloud Service account information.
+///
+/// The [agentId] has the following format: `service:gcp-service-account:user@examplec.com`
 class AuthenticatedGcpServiceAccount implements AuthenticatedAgent {
   @override
-  String get agentId => KnownAgents.gcpServiceAccount;
+  late final agentId = [
+    KnownAgents.gcpServiceAccount,
+    email,
+  ].join(':');
```

Comment:
hmm, we could consider using `oauthUserId`.

It's less readable, and not very nice. So I'm inclined to also go with email.

We could also use `repository_id` in the github agent id. Then it would be the same identifier across renames, but it's also a bit crazy -- and impossible to read.

On the flip side: the UUID used for user_id isn't very readable either.

I guess I'm wondering what is it we want agentId to be?

--------------

We should probably also update the documentation `AuthenticatedAgent.agentId` regardless.

Reply:
> We could also use repository_id in the github agent id. Then it would be the same identifier across renames, but it's also a bit crazy -- and impossible to read.

We are allowing the publishing based on the `<org>/<repo>` name (as stored in the package `AutomatedPublishing` object), so it may seem better to use that as the agent id. We could store the ids in the attached `data` map though

.> On the flip side: the UUID used for user_id isn't very readable either.

Yeah, but at least that is the only UUID we are using and the lookup to email is straightforward. So there is very little benefit of changing that now.

Reply:
Yeah, I'm undecided on this too. It may be better to use a more opaque id, especially if we know that it won't change in the future. `repositoryId` may change: people may delete their repository and recreate it, using the same name, but resulting in a different id. It is less likely with the oauthUserId. So maybe use that here?

Reply:
Crazy idea; what about `<org>/<repo>/<repoId>` and similarly `<email>/<oauthUserId>`?